### PR TITLE
fix: wallet accounts ordering improvements

### PR DESCRIPTION
**Pull request by saledjenic**
Some improvements done for maintaining accounts' positions.

Instead of the old way which allowed gaps in position, now we don't have gaps, but moving certain account from position to position.

Also for the purpose of accounts order syncing we're sending all known accounts (all keypairs' accounts, including watch only accounts, but without status chat account which doesn't participate in wallet section) as a new `protobuf.SyncAccountsPositions` message.

If anybody is interested in the functionality, it can be seen in the video attached in the description of this PR: https://github.com/status-im/status-desktop/pull/11556

### CODE REVIEW

File: appdatabase/migrations/sql/1689498471_make_wallet_accounts_positions_non_negative.up.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	chat = TRUE;
+
+UPDATE
+	keypairs_accounts AS ka
+SET
+	position = kat.pos
+FROM (
+    SELECT
+		ka.name,
+		ROW_NUMBER() OVER (ORDER BY kat.position) - 1 AS pos
```
Comment (qfrank) [inline appdatabase/migrations/sql/1689498471_make_wallet_accounts_positions_non_negative.up.sql]: proberbly we'll face [like issue](https://github.com/status-im/status-mobile/issues/16421) again if we use `ROW_NUMBER()` , [relate fix](https://github.com/status-im/status-go/pull/3687)

Comment (saledjenic) [reply appdatabase/migrations/sql/1689498471_make_wallet_accounts_positions_non_negative.up.sql]: Updated.

File: protocol/messenger.go [hunk 2]
```diff
@@ -2601,6 +2601,11 @@ func (m *Messenger) SyncDevices(ctx context.Context, ensName, photoPath string,
 		return err
 	}
 
+	err = m.syncAccountsPositions(rawMessageHandler)
```
Comment (qfrank) [inline protocol/messenger.go:2604]: I haven't see corresponding changes for [HandleSyncRawMessages](https://github.com/status-im/status-go/blob/f806a0b8cfde8cb4873edccf4f7ea0d0eb07615a/protocol/messenger_sync_raw_messages.go#L19) ?

File: multiaccounts/accounts/database.go [hunk 3]
```diff
@@ -1146,41 +1168,137 @@ func (db *Database) UpdateAccountPosition(address types.Address, newPosition int
...
+	// returns all accounts ordered by position
+	dbAccounts, err := db.getAccounts(tx, types.Address{})
 	if err != nil {
 		return err
 	}
-	acc, err := db.getAccountByAddress(tx, address)
-	if err != nil {
-		return err
+
+	// starting form -1, cause `getAccounts` returns chat account as well
```
Comment (qfrank) [inline multiaccounts/accounts/database.go]: form -> from

**Comment by saledjenic** [comment]
@qfrank thanks, updated.

**Review by qfrank — APPROVED** [review]

**Review by alaibe — APPROVED** [review]

**Review by alaibe — APPROVED** [review]

**Pull request merged**